### PR TITLE
Deterministic stdenv

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10304,6 +10304,9 @@ in
   gcc9Stdenv = overrideCC gccStdenv buildPackages.gcc9;
   gcc10Stdenv = overrideCC gccStdenv buildPackages.gcc10;
 
+  # This is not indented for packages in nixpkgs, but for providing a
```
Comment (SFrijters) [inline pkgs/top-level/all-packages.nix:10307]: "indented" -> "intended"?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10304,6 +10304,9 @@ in
...
+  # faster-running compiler to nixpkgs user by enabling profile-guided
+  # optimizations options when building gcc.
```
Comment (Atemu) [inline pkgs/top-level/all-packages.nix:10309]:
```suggestion
  # This is not intented for use in nixpkgs but for providing a faster-running
  # compiler to nixpkgs users by building gcc with reproducibility-breaking
  # profile-guided optimizations
```

**Review by Atemu — CHANGES_REQUESTED** [review]
Maybe the intended use should also be documented in `fastStdenv.meta.description` while we're at it, not sure how trivial that would be.

**Review by Atemu — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -10304,6 +10304,9 @@ in
   gcc9Stdenv = overrideCC gccStdenv buildPackages.gcc9;
   gcc10Stdenv = overrideCC gccStdenv buildPackages.gcc10;
 
+  # This is not intented for use in nixpkgs but for providing a faster-running
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10307]:
```suggestion
  # This is not intended for use in nixpkgs but for providing a faster-running
```

Comment (Atemu) [reply pkgs/top-level/all-packages.nix:10307]: Damnit xD

**Review by tomberek — APPROVED** [review]

**Pull request merged**